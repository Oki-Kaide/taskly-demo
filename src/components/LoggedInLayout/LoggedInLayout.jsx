import React from 'react';
import PropTypes from 'prop-types';
import { RedBackground } from '../index';
import './LoggedInLayout.sass';

class LoggedInLayout extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      windowWidth: 0,
    };
  }

  componentDidMount() {
    this.updateWindowWidth();
    window.addEventListener('resize', this.updateWindowWidth);
  }

  updateWindowWidth = () => {
    this.setState({ windowWidth: window.innerWidth });
  }

  render() {
    const { children, mainImageUrl, avatar, logout } = this.props;
    const { windowWidth } = this.state;

    return (
      <div className="page">
        <div className="page-wrapper">
          <div className="header">
            {windowWidth > 700 ? (
              <img src="./images/taskly-logo-white.svg" alt="Taskly" />
            ) : (
              <img src="./images/taskly-logo.svg" alt="Taskly" />
            )}
            <div
              style={{
                backgroundImage:
                  avatar !== ""
                    ? `url('data:image/jpeg;base64,${avatar}')`
                    : `url('./images/default-avatar.png')`,
              }}
              alt="avatar"
              className="header-avatar"
              onClick={logout}
            />
          </div>
          <div className="content-wrapper">
            <div className="content">
              {children}
              <RedBackground side="left" bottom="-5px">
                <img
                  src={mainImageUrl}
                  alt="Smiling"
                  className="account-image"
                />
              </RedBackground>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default LoggedInLayout;

LoggedInLayout.propTypes = {
  avatar: PropTypes.string,
  openConfirmModal: PropTypes.func,
  logout: PropTypes.func.isRequired,
};

LoggedInLayout.defaultProps = {
  avatar: '',
  openConfirmModal: null,
};
