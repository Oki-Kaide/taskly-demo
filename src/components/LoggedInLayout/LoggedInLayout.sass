@import '../../styles/variables.sass'
@import '../../styles/mixins.sass'

.content-wrapper,
.content
  display: flex
  flex-direction: column

.content-wrapper
  @include for-tablet-portrait-up
    position: fixed
    top: -20px
    width: calc(100vw - 120px)
    max-width: 1218px
    align-items: flex-end
    height: 100vh
  @include for-phone-only
    margin-top: 10px
    padding-bottom: 30px

.content
  justify-content: center
  height: 100%
  @include for-tablet-portrait-landscape-up
    width: 332px
  @include for-tablet-landscape-up
    width: 400px

  h1
    margin-bottom: 10px