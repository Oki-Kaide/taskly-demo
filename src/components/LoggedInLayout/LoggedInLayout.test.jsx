import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import LoggedInLayout from './LoggedInLayout';

const createTestProps = props => ({
  logout: jest.fn(),
  children: <div />,
  mainImageUrl: './images/smiling-girl-bust.png',
  avatar: '',
  ...props,
});

describe('LoggedInLayout component', () => {
  let props;

  beforeEach(() => {
    props = createTestProps({});
  });

  it('renders without crashing', () => {
    shallow(<LoggedInLayout {...props} />);
  });

  it('renders without crashing with non empty avatar', () => {
    shallow(<LoggedInLayout {...props} avatar='hi' />);
  });

  it('matches the rendered snapshot if windowWidth is less than 700', () => {
    const component = shallow(<LoggedInLayout {...props} />);
    component.setState({ windowWidth: 600 });
    expect(toJson(component)).toMatchSnapshot();
  });

  it('matches the rendered snapshot if windowWidth is greater than 700', () => {
    const component = shallow(<LoggedInLayout {...props} />);
    component.setState({ windowWidth: 800 });
    expect(toJson(component)).toMatchSnapshot();
  });
});
