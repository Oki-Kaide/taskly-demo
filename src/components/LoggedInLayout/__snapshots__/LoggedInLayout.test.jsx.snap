// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoggedInLayout component matches the rendered snapshot if windowWidth is greater than 700 1`] = `
<div
  className="page"
>
  <div
    className="page-wrapper"
  >
    <div
      className="header"
    >
      <img
        alt="Taskly"
        src="./images/taskly-logo-white.svg"
      />
      <div
        alt="avatar"
        className="header-avatar"
        onClick={[MockFunction]}
        style={
          Object {
            "backgroundImage": "url('./images/default-avatar.png')",
          }
        }
      />
    </div>
    <div
      className="content-wrapper"
    >
      <div
        className="content"
      >
        <div />
        <RedBackground
          bottom="-5px"
          left="auto"
          right="auto"
          side="left"
          top="auto"
        >
          <img
            alt="Smiling"
            className="account-image"
            src="./images/smiling-girl-bust.png"
          />
        </RedBackground>
      </div>
    </div>
  </div>
</div>
`;

exports[`LoggedInLayout component matches the rendered snapshot if windowWidth is less than 700 1`] = `
<div
  className="page"
>
  <div
    className="page-wrapper"
  >
    <div
      className="header"
    >
      <img
        alt="Taskly"
        src="./images/taskly-logo.svg"
      />
      <div
        alt="avatar"
        className="header-avatar"
        onClick={[MockFunction]}
        style={
          Object {
            "backgroundImage": "url('./images/default-avatar.png')",
          }
        }
      />
    </div>
    <div
      className="content-wrapper"
    >
      <div
        className="content"
      >
        <div />
        <RedBackground
          bottom="-5px"
          left="auto"
          right="auto"
          side="left"
          top="auto"
        >
          <img
            alt="Smiling"
            className="account-image"
            src="./images/smiling-girl-bust.png"
          />
        </RedBackground>
      </div>
    </div>
  </div>
</div>
`;
